provider/app.py

```diff
@@ -158,19 +164,20 @@ def authorizedForTrigger(auth, consumer):
 
 
 def createAndRunConsumer(triggerFQN, params, record=True):
-    if app.config['TESTING'] == True:
-        logging.debug("Just testing")
-    else:
-        # generate a random uuid for new triggers
-        if not 'uuid' in params:
-            params['uuid'] = str(uuid.uuid4())
+    # generate a random uuid for new triggers
+    if not 'uuid' in params:
+        params['uuid'] = str(uuid.uuid4())
 
+    if 'status' not in params or params['status']['active'] == True:
+        logging.info('{} Trigger was determined to be active, starting...'.format(triggerFQN))
         consumer = Consumer(triggerFQN, params)
         consumer.start()
         consumers.addConsumerForTrigger(triggerFQN, consumer)
+    else:
+        logging.info('{} Trigger was determined to be disabled, ignoring...'.format(triggerFQN))
```

Comment:
During provider boot up, this function is called for all existing triggers in the DB with the `record` flag False. That is, if the trigger already exists in the DB (currently all disabled triggers are already in the DB as this function is never called to create a new disabled trigger), then this code will not write that document again.

Conversely, you _could_ call this function as-is to create a new disabled trigger, in which case you would want to record it to the DB. Although this will never happen, the current code would correctly handle it.